src/easylink/step.py

```diff
@@ -815,6 +815,27 @@ def set_configuration_state(
             parent_config, combined_implementations, input_data_config
         )
 
+    def _duplicate_template_step(self) -> Step:
+        """Makes a duplicate of the template ``Step``.
+
+        A duplicate of the :attr:`template_step`, i.e. the ``Step`` that may have
+        multiplicity, should contain copies of every attribute of said ``Step``
+        *except* for the :attr:`parent_step`. This is because :attr:`parent_step`
+        contains an actual ``Step`` instance and we want all of the duplicates to
+        have that same parent as opposed to copies of one.
+
+        Returns
+        -------
+            A duplicate of the :attr:`templated_step`.
+        """
+        # Due to the cyclic nature, self == self.template_step.parent_step
+        parent_step = self
+        self.template_step.parent_step = None
+        step_copy = copy.deepcopy(self.template_step)
+        step_copy.set_parent_step(parent_step)
+        self.template_step.set_parent_step(parent_step)
```

Comment:
This wasn't previously being done, but I'm actually kinda surprised it didn't break things. It just seems more proper to re-set the parent on the template.